blog/.vuepress/theme/Layout.vue

```diff
@@ -21,12 +26,30 @@ import SiteNavigation from './partials/Navigation'
 
 export default {
   components: { Page, Posts, Post, SiteFooter, SiteHeader, SiteNavigation },
+  data: function () {
+    return {
+      fixedNav: ''
+    }
+  },
+  created () {
+    window.addEventListener('scroll', this.handleScroll);
+  },
+  destroyed () {
+    window.removeEventListener('scroll', this.handleScroll);
+  },
   methods: {
     ...mapActions(['updateSite', 'updatePage', 'updateParams']),
-      updateLayoutClass () {
-        this.$el.parentNode.className = `${this.type}-template`
-      }
+    updateLayoutClass () {
+      this.$el.parentNode.className = `${this.type}-template`
     },
+    handleScroll: function () {
+      var scroll = window.scrollY;
+      var headerElement = document.getElementById('site-header');
```

Comment:
you can use the vue native ref functionality (https://vuejs.org/v2/api/#ref). Executing document selectors in a Vue instance is not a good pattern since you run outside of the vue component lifecycle.

Reply:
This is only possible in `Layout.vue`. I `Header.vue` it is not really possible as I reference an element from a different component. So I have to stick with `document.getElementsByClassName("post-full-title")[0];` there. Or do you have a good idea for this?